gammapy/obs/tests/test_observation.py

```diff
@@ -9,3 +14,181 @@
 
 def test_ObservationTable():
     ObservationTable()
+
+
+def common_sky_region_select_test_routines(obs_table, selection):
+    """Common routines for the tests of sky_box/sky_circle selection of obs tables"""
+    type = selection['type']
+    if type not in ['sky_box', 'sky_circle']:
+        raise ValueError("Invalid type: {}".format(type))
+
+    if type == 'sky_box':
+        lon_range_eff = (selection['lon'][0] - selection['border'], selection['lon'][1] + selection['border'])
+        lat_range_eff = (selection['lat'][0] - selection['border'], selection['lat'][1] + selection['border'])
+    elif type == 'sky_circle':
+        lon_cen = selection['lon']
+        lat_cen = selection['lat']
+        center = SkyCoord(lon_cen, lat_cen, frame=selection['frame'])
+        radius_eff = selection['radius'] + selection['border']
+
+    do_wrapping = False
+    # not needed in the case of sky_circle
+    if (type == 'sky_box' and
+        any(l < Angle(0., 'degree') for l in lon_range_eff)):
+        do_wrapping = True
+
+    # observation table
+    skycoord = skycoord_from_table(obs_table)
+
+    # test on the selection
+    selected_obs_table = obs_table.select_observations(selection)
+    skycoord = skycoord_from_table(selected_obs_table)
+    if type == 'sky_box':
+        skycoord = skycoord.transform_to(selection['frame'])
+        lon = skycoord.data.lon
+        lat = skycoord.data.lat
+        if do_wrapping:
+            lon = lon.wrap_at(Angle(180, 'degree'))
+        assert ((lon_range_eff[0] < lon) & (lon < lon_range_eff[1]) &
+                (lat_range_eff[0] < lat) & (lat < lat_range_eff[1])).all()
+    elif type == 'sky_circle':
+        ang_distance = skycoord.separation(center)
+        assert (ang_distance < radius_eff).all()
+
+    # test on the inverted selection
+    selection['inverted'] = True
+    inv_selected_obs_table = obs_table.select_observations(selection)
+    skycoord = skycoord_from_table(inv_selected_obs_table)
+    if type == 'sky_box':
+        skycoord = skycoord.transform_to(selection['frame'])
+        lon = skycoord.data.lon
+        lat = skycoord.data.lat
+        if do_wrapping:
+            lon = lon.wrap_at(Angle(180, 'degree'))
+        assert ((lon_range_eff[0] >= lon) | (lon >= lon_range_eff[1]) |
+                (lat_range_eff[0] >= lat) | (lat >= lat_range_eff[1])).all()
+    elif type == 'sky_circle':
+        ang_distance = skycoord.separation(center)
+        assert (ang_distance >= radius_eff).all()
+
+    # the sum of number of entries in both selections should be the total number of entries
+    assert len(selected_obs_table) + len(inv_selected_obs_table) == len(obs_table)
+
+
+def test_select_parameter_box():
+    # create random observation table
+    observatory_name='HESS'
```

Comment:
The `observatory_name` is irrelevant for this test, so we should not have to give it.
Does this work? (if no, can you make this work?)
```python
 obs_table = make_test_observation_table(n_obs=10)
```